### ISSUE TITLE
Temperatures computed and LWT temperatures

### ISSUE THREAD

**Issue by ogkita** [ysard/ebusd_configuration_chaffoteaux_bridgenet#7]
Hola @ysard, @wrongisthenewright.

This topic for computation temperatures discursion ;-)

Yes, I have read the installer document, about 20 or 25 times... I even generated the translated version from the Italian... It helped a bit, but do not think it helped a lot.

In the document, it explains how it works in heating mode. The formula... understood and there are no doubts. I hope to be able to check the values it calculates and uses next winter.

Now, in cooling mode I have not found in the document the formula that applies. I understand that it cannot be the same as the slope configured has nothing to do, in the case of heating it goes from 0.2 to 1 and in cooling it goes from 0 to 30. They are very different values and I think that the same formula does not support these variations:

   For heating:
![image](https://github.com/user-attachments/assets/17169e4a-6a8a-47ee-93ba-0facc90177f6)

   For cooling:
![image](https://github.com/user-attachments/assets/dd95c3a6-862a-43e5-9609-f3507c939cca)

For heating. Simplifying and using only external probe:

Tset_term = Tmand_min +SL*( Tamb_set -Text)+OF.
dove:
- Tmand_min = lwt_min, parameters 6171 6172 6173. We will use the same value in all. 25°C
- Tamb_set = Target room temperature, parameters 6271, 6272, 6273. We will use the same value in all of them. 21°C
- Text = External temperature. param 7647. Here we have 2 parameters 7647 external probe and 7810 from the hp, but I understand that the system uses the external probe. -5-20:

![image](https://github.com/user-attachments/assets/a10d0078-87ed-4d03-9268-3ad7cbac2175)

II must wait to winter to check if this is correct.

For cooling. Remember that I have not found the formula that applies in the document you indicate. Just changing the external temperature and the slope in the above formula we have:

![image](https://github.com/user-attachments/assets/095efc0b-e80a-4143-9f6b-a7c0c9e1c714)

I think it is not correct

  The algorithm is influenced by various offset that must be taken into account when checking the temps.
  There is a heating (par 12.2.a1) and a cooling (par 12.4.2) compensation offset for LWT at HP level, this helps to deliver the right water temp at the zone manager in case the water gets cooled (in heating mode) or heated (in cooling mode) while the water is in transit from the HP to the zone manager for poor insulation or very long tubing.

It may be as you indicate, the point is that to simplify everything I have the offset parameters set to 0.

  7f2b
  7571
  7572
  7573

In addition, I have checked the following values in the menu and they are all set to 0:

  For heating, these should not affect the cooling system:
  4.2.3
  4.2.4
  5.2.3
  5.2.4
  6.2.3
  6.2.4

  For cooling, these are now the ones that should be affected:
  4.5.4
  5.5.4
  5.5.4

  7.3.0

  Remember I configured this params at 0,

Finally, I believe that by simply applying your formula and making the temperature delivery fixed, we are losing some feature of the system.....

Another thing I have done is to ask different AI if they would give me a formula for heating and a formula for cooling using this data:

  Current:
    External temperature.
    External humidity.
    Wind speed.

  Forecast 3h:
    Outside temperature.
    External humidity.
    Wind speed.

  Room:
    Current temperature.
    Target temperature.

  Factors:
    Housing insulation level: 0-10, in order to be able to set several.
    Outside influence on the calculation: 0-10, so that several can be configured.
    Room influence on the calculation: 0-10, so that several can be configured.

Let's say that this could be as complicated as we want and the more data we have, the more optimal it will be...

Well, sorry for the talk... We are still working on it!

Thanks, Oscar

**Comment by wrongisthenewright** [comment]
> This topic for computation temperatures discursion ;-)

> Now, in cooling mode I have not found in the document the formula that applies. I understand that it cannot be the same as the slope configured has nothing to do, in the case of heating it goes from 0.2 to 1 and in cooling it goes from 0 to 30. They are very different values and I think that the same formula does not support these variations:

I was wrong, you're right, the cooling curves are different from heating.

these came from Italian manual low temp (fancoils) and high temp (underfloor cooling)

![image](https://github.com/user-attachments/assets/9ab4c1bf-6120-4114-8fc8-4ededc10a263)

It's obviously an inversely proportional slope with imposed limits (limits btw that are function of the chosen slope) , the formula should be:

Tset_cooling = T_mand_min + SL/100*(40°C- Text)+OF.

So if you're on underfloor config, with a min temp of 18, no offset, slope 5 and ext temp of 30°C the value is:
Tset_cooling = 18°C + 5/100*(40°C - 30°C) + 0   = 18.5°C

in case of fancoil config, with a min temp of 7, no offset, slope 25 and ext temp of 30°C the value is:
Tset_cooling = 7°C + 25/100*(40°C - 30°C) + 0   = 9.5°C

Both values matches the graph.

> It may be as you indicate, the point is that to simplify everything I have the offset parameters set to 0.

OK, this simplify the calculations, but the offset are there for a purpose, so you'll need to take them into account if you'll need to use them.

> Finally, I believe that by simply applying your formula and making the temperature delivery fixed, we are losing some feature of the system.....

Choosing the right algorithm is not a matter of losing features, is mostly related to cost/savings and overall system configuration, in my case, given my needs and my home configuration I use a different configuration for cooling and heating.

In winter (heating) I use full automatic calculations with external and ambient probe. Having an hibrid system, once the energy manager has all the parameters chooses the LWT and then chooses to use either the boiler, the HP or both to get that temp, all based on the energy tariffs I inserted. In this way I always get the max savings in every instant.

In summer (cooling) I use a fixed temp configuration at the lower temp possible (7°C). If I'd have chosen the automatic calculations I may have had some extra saving because the HP could have run at lower frequencies (HP tend to perform better in termo of COP/EER in the middle range of functioning frequencies), but I don't mind losing some KWh here and there because I have a PV and battery systems at home , so in summer I get my electric energy for free from the sun. 

> Another thing I have done is to ask different AI if they would give me a formula for heating and a formula for cooling using this data:

Interesting, but let's also take into account that the heating system of a home is one of it's core features. Commercial HVAC products may be overpriced and lacking some "finesse" but I wouldn't leave the comfort of my home in the and of a DIY algorithm that may run on an off the shelf HW that breaks down in the middle of the winter because some nerd put out the wrong update.

I'm into home automation allright and my being here it's a proof of that, but my heating is solely managed by the Ariston thermostat, the calculations made by the machine and my only intervention (apart from gathering metrics for reporting) is limited to change energy tariffs based on my contracts and PV production in winter.

**Comment by wrongisthenewright** [comment]
Hello Oscar,

> Finally, I believe that by simply applying your formula and making the temperature delivery fixed, we are losing some feature of the system.....

Sorry I was pondering on other topics and came back here...if you're referring to losing feature of the system by forcing it to accept you externally calculated temps I don't think you'll lose anything.
If the system accept your calculated temp (you can check it in par 17.14.1, I think) then every other aspect is derived from there and, obviously, from the other parameters like tipe of modulation for the pump, silent mode etc.

If the system is forced (by your writing messages) to reach a target temp that you send, then the machine set the compressor frequency to go there, the pump modulation follows along etc.

As I wrote before I don't think it's wise to do so, in my house I'd prefer to have the sensys thermostats for all zones placed in the appropriate rooms (the colder in winter and/or hotter in summer) and start from there, then I'd put in the various rooms without sensys some mechanism to shut off the heating/cooling for that room if need be, as an exampre closing an actuated valve or shutting off the relative fancoil, but that with a phisical, dedicated and purpose built thermostat, not an ESP with a temp sensor.
In this way the system can run autonomously, without fear of losing something for a bad firmware upgrade.

**Comment by ogkita** [comment]
Hi @wrongisthenewright

  Sorry I was pondering on other topics and came back here...

Don't worry, I got off to a bad start with the topics ;-)

  Tset_cooling = T_mand_min + SL/100*(40°C- Text)+OF.
  
  So if you're on underfloor config, with a min temp of 18, no offset, slope 5 and ext temp of 30°C the value is:
  Tset_cooling = 18°C + 5/100*(40°C - 30°C) + 0 = 18.5°C
  
  in case of fancoil config, with a min temp of 7, no offset, slope 25 and ext temp of 30°C the value is:
  Tset_cooling = 7°C + 25/100*(40°C - 30°C) + 0 = 9.5°C

Thank you for clarifying the cooling formula. Now  I have 3 external temperatures (weather, external probe and hp), for T_mand_min = 18, your formula result is:

27,8  18,60
28,5  18,60
29,5  18,50

and I can see the computed temperature is in all zones 18.6 °C. Makes sense.

I still don't understand why the setpoint in the heat pump is 16.6, as I have all offsets at 0, or so I think. I have also noticed that this value changes if one zone or several zones are working, it must apply some "factor" to avoid that if there is a lot of demand the system is not efficient...

  Sorry I was pondering on other topics and came back here...if you're referring to losing feature of the system by forcing it to accept you externally calculated temps I don't think you'll lose anything.
  If the system accept your calculated temp (you can check it in par 17.14.1, I think) then every other aspect is derived from there and, obviously, from the other parameters like tipe of modulation for the pump, silent mode etc.
  
  If the system is forced (by your writing messages) to reach a target temp that you send, then the machine set the compressor frequency to go there, the pump modulation follows along etc.

I will start by applying the algorithm to the fixed temperature and see how it differs from the Ariston computed.

  As I wrote before I don't think it's wise to do so, in my house I'd prefer to have the sensys thermostats for all zones placed in the appropriate rooms (the colder in winter and/or hotter in summer) and start from there, then I'd put in the various rooms without sensys some mechanism to shut off the heating/cooling for that room if need be, as an exampre closing an actuated valve or shutting off the relative fancoil, but that with a phisical, dedicated and purpose built thermostat, not an ESP with a temp sensor.
  In this way the system can run autonomously, without fear of losing something for a bad firmware upgrade.

The problem is my installation. I have:

Zone 1:
  Thermostat 1, commands over TA1
Zone 2:
  Thermostat 2, commands over TA2
  Thermostat 3, commands over TA2
Zone 2:
  Thermostat 4, commands over TA3
  Thermostat 5, commands over TA3
  Thermostat 6, commands over TA3

Ariston thermostats only support one thermostat over 1 zone, in my installation I need several thermostat over 1 zone... 

Please don't ask me why this is so, the installer let me know at the last minute and I had to find my own way to get something decent. ....

I have recently contacted Ariston to see if they have "solved" the problem and I am still waiting for a reply. I don't have much hope for it.

**Comment by ysard** [comment]
> In this way the system can run autonomously, without fear of losing something for a bad firmware upgrade.

Well, you know, there are a few (BTW rare and underpaid) people who know how to develop tested and reliable firmwares :grinning:

**Comment by wrongisthenewright** [comment]
> > In this way the system can run autonomously, without fear of losing something for a bad firmware upgrade.
> 
> Well, you know, there are a few (BTW rare and underpaid) people who know how to develop tested and reliable firmwares 😀 😆

In Italian we say “presenti esclusi” that means something like I was speaking in general and not referring to someone “present” here :-D

**Comment by ogkita** [comment]
Yes, everything has its advantages and disadvantages... Having everything integrated in one place means that you can manage everything from one app, all "things" talk to others.

The issue in updates... it doesn't have to fail, but tell that to MS... ;-). In any case, having a good backup/restore plan and doing only critical or tested updates is a good option. Here we say "if something is working do not touch" ;-)

I like to have always a plan B or C with less "domotic" or even manual.

**Comment by ogkita** [comment]
Regarding temperature computing... ;-)

Things I have seen in cooling mode that I think also applies to heating (zone 1 parameters only and applies to all).

When you use thermoregulation 1, fixed temperature.

You set parameter 7071 with the temperature you want (using whatever data to calculate it) and the system automatically sets that value in parameter 6197.

When using thermoregulation 2, External probe

The system ignores parameter 7071 and automatically calculates and sets that value in parameter 6197.

In other words, in the end the system has the values in the parameters 6197 6297 6397 and with these it calculates the parameter 6047.

Computation of parameter 6047.

1.- 6047 = The minimum value of 6197 6297 6397.
2 - Apply configuration of menu 7.3.0.
   2.1- If 7.3.0>0.
      2.1.1 Fancoils. 6047 = 6047 - menu 7.2.1 (I think this is parameter 7f2b).
      2.1.2 floor cooling. 6047 = 6047 - 2 - menu 7.2.1 (I think this is the parameter 7f2b)
   2.2.- If 7.3.0=0
      2.2.1 Applies an automatic "computation" based on the number of zones are demanding and the type of zone management module (MGZ).

Now I research what happen with LWT, I thing this value is 6047 param with some computation..., maybe the system is using here delta_T???

> If the system accept your calculated temp (you can check it in par 17.14.1, I think) then every other aspect is derived from there and, obviously, from the other parameters like tipe of modulation for the pump, silent mode etc.

You are right, it seems to be all about the XXX parameter 6047...

**Comment by wrongisthenewright** [comment]
> Regarding temperature computing... ;-)

> You set parameter 7071 with the temperature you want (using whatever data to calculate it) and the system automatically sets that value in parameter 6197.

I agree on this,  similar approach should be detected on heating, only 7071 became 6571. I haven't made any test so I don't know if the system applies to this value the HP "general" offset and the specific zone offset.

> When using thermoregulation 2, External probe 

Or External+Ambient probe for that matter

> The system ignores parameter 7071 and automatically calculates and sets that value in parameter 6197.

Yes, it calculates 6197 as the minimum of the various zones, in heating the reverse is true (the maximum of 6571,6572, 6573)

> Computation of parameter 6047.

I think the "formula" is like this:

6047 = min (6197 +/- 7571, 6297 +/- 7572, 6397 +/- 7573) +/- 7f2b

calculate the target for each zone, applies the specific offset, detect the minimum for all zones then applies the "general" offset.

> Now I research what happen with LWT, I thing this value is 6047 param with some computation..., maybe the system is using here delta_T???

LWT is not calculated. It is DETECTED when the water flows back. The machines then calculate the difference from 6047 and LWT temp, if it is above the delta_t configured in 12.3.0 (in my system, be aware that different configuration get different menu numbering...., in your case it may be 17.3.0 or 7.3.0) then the HP runs ah hingher power to reach the requested delta.

**Comment by wrongisthenewright** [comment]
Only to report that I think I've found the reasons behind the 3 different ext temps:

the HP internal probe is mandatory, it is used to calculate when to do the defrost in heating mode. Par. 16.9 of the professional manual shows the algorithm; given that in some installation the HP can be used without internet and without external probe this is the only way the machine can calculate when do the defrost

external probe, if present is used to do the target temp computation, but, i think, not the defrost

in apsence of external probe internet weather temp is used to do the target temp calculation, also in this case i think it's not used to decide on defrost (eg. resource temporarily unavailabe/ false reading)

**Comment by ogkita** [comment]
Hi,

I am sorry for the silence all this time but I had another home project with OCPP for the car charger....

In all this time I have been able to verify that the system works as discussed here: Basically you set the setpoint per zone (either calculated by the machine or by the domotics) and the machine modulates depending on what is needed.

As much as I investigate, I am not able to see what are the “final” offsets that the system applies to calculate the set_point of the hp. From what I have seen in heating mode it always adds 5 degrees for each zone that is demanded and keeps those 5 degrees offset if there is no demand.

I have also noticed that every x minutes the set_point sets it to 8 degrees... I don't see the point unless it is some anti-freeze safety system or something the machine has.

![Image](https://github.com/user-attachments/assets/efe186e1-a0fb-4c91-92b3-8005bbdc2789)

@wrongisthenewright , I read on another forum that you were having problems with random reboots. I have had them too and it seems to be due to scheduled ebusctl connections (e.g. cron). I removed these connections and the “questions” are done with mqtt and the random restarts have disappeared.

I don't know if you remember that I contacted Ariston to see if it was possible to install several thermostats of their brand in the same zone. By email they replied that it was possible, they came to my house and explained that it was not possible.... A bit disaster.

Now I am working on the following:

	1.- Establish the fixed setpoint by doing the calculation in domotic. DONE. Except for that offset that I can't control, but I can live with it.

	2.- When the room temperature is reached, instead of turning off the relay of the zone, what I want is to change the fixed setpoint to the temperature that corresponds to the machine to continue working always but with more modulation. In this way I hope to have less ON/OFF and the same comfort.

		2.1- Calculate this new setpoint for each zone, only case target_temp_ZX = current_temp_ZX

		2.2- OFF rele ZX, only when target_temp_ZX < (current_temp_ZX + margen_temp_ZX). I am working now with margen_temp_ZX = 0,5º.

I will back with news... I hope soon ;-)
, Oscar

**Comment by ysard** [comment]
Hum, the configured minimum temperature seems to prevent the set temperature from going below (~35°C from what I see).
The change to 8°C in frost protection mode seems to correspond to the scheduled heating shutdowns (OFF).
The fact that the time spent in this mode is short is in my opinion due to the fact that you have 3 zones likely to request
a restart at any time as soon as the demand exceeds the set set-point by 0.5°C.
The boiler is indeed unlikely to stop for a long time.

You should be interested in the overall heating time (cumulative time spent in "heating" mode computed by a script for example).
For my part, a heating restart results
in a heating time equivalent to approximately 1h30 of operation
(excluding any consideration of heating power involved => I do not have this data on the Mira C Green).
Consequently, you must code a hysteresis in both temperature AND duration.
Ex: a shutdown can only be followed by a restart if X minutes have elapsed.
